client/src/components/busseat.js

```diff
@@ -1,12 +1,272 @@
+  const [bus,setBus]=useState([]);
+  const [seatPrice,setSeatPrice]=useState([])
+  const [arrivalTime,setArrivalTime]=useState([])
+  const [departureTime,setDepartureTime]=useState([])
+
+  const [name, setName] = useState([])
+  const [gender, setGender] = useState([])
+  const [booked,setBookedSeats]=useState([])
+  const [image,setImage]=useState(['../../images/available.png'])
+
+  const {id,doj}=useParams()
+  console.log(id)
+  console.log(doj)
+
+  const javascript_date = new Date(doj);
+  console.log(javascript_date)
+  const day=javascript_date.getDay()
+ // console.log(getweekday(day))
+
+
+
+  // const queryParameters = new URLSearchParams(window.location.search)
+  // const b_id = queryParameters.get("id")
+  // const dateOfJourney = queryParameters.get("doj")
+  // console.log(b_id)
+  // console.log(dateOfJourney)
+
+  const getweekday=(day)=>{
+    if(day==0){
+      return 'sunday'
+    }
+    else if(day==1){
+      return 'monday'
+    }
+    else if(day==2){
+      return 'tuesday'
+    }
+    else if(day==3){
+      return 'wednesday'
+    }
+    else if(day==4){
+      return 'thursday'
+    }
+    else if(day==5){
+      return 'friday'
+    }
+    else{
+      return 'saturday'
+    }
+  }
+  
+
+  const handleSeatSelection = (seat) => {
+    if (!selectedSeats.includes(seat)) {
+      setSelectedSeats([...selectedSeats, seat]);
+    } else {
+      setSelectedSeats(selectedSeats.filter(s => s !== seat));
+    }
+  };
+
+
+  const hanledisable=(seat)=>{
+    if(booked.includes(seat)){
+      return true
+    }
+    else if(ignore.includes(seat)){
+      return true
+    }
+
+  }
+
+  const handleStyle=(seat)=>{
+    if(ignore.includes(seat)){
+      return {width: "80px",height:"80px",visibility:'hidden'}
+    }
+  }
+
+  const handlevariant=(seat)=>{
+    if(ignore.includes(seat)){
+      return 'outline-light'
+    }
+    if(booked.includes(seat)){
+      return 'outline-light'
+    }
+    return selectedSeats.includes(seat)? 'outline-green' : 'outline-light';
+  }
+
+
+  const [ticket, setTicket] = useState('')
+  const navigate = useNavigate()
+  const handleBookNowClick =async()=>{
+    try{
+      console.log(getweekday(day));
+      // const post_data = {
+      //   selected_seats: selectedSeats,
+      //   date: doj,
+      //   bus_id:bus["_id"],
+      //   user_id:"1",
+      //   ticket_price:bus["ticket_price"],
+      //   day:getweekday(day),
+      // }
+      const post_data = {
+        selected_seats:selectedSeats ,
+        date: doj,
+        bus_id:bus["_id"],
+        user_id:"63e49ceca788d71cb4dae60c",
+        ticket_price:seatPrice * selectedSeats.length,
+        day:`${getweekday(day)}`,
+      }
+      const response = await fetch('http://127.0.0.1:4000/ticket', {
+        method: 'POST', 
+        mode:'cors',
+        headers: {
+          'Content-Type': 'application/json; charset=UTF-8'
+        },
+        body: JSON.stringify(post_data)
+      });
```

Comment:
Refactor into another file, say `api.js` that contains all such function calls to the backend

Reply:
Comment applies to all such `fetch` calls